GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace GenericsHomework;
+public class Node<T>
+{
+    private int Size { get; set; }
+    public T? Item { get; set; }
+    public Node<T> Next
+    {get; private set; }
+
+    public Node(T? item, Node<T> next)
+    {
+        Item = item;
+        Next = next ?? this;
+        Size = 1;
+    }
+
+    public override string ToString()
```

Comment:
You can return a nullable string here, which might simplify the below code
```suggestion
    public override string? ToString()
```